### DISCUSSION TITLE
Possible error that may occur

### DISCUSSION THREAD

**Discussion by zhangxin702** [Autostronomy/AstroPhot#107] [General]
Hello, I have noticed that when I call the function gaussian_torch(R, sigma, I0), I often encounter the following error: "RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!" However, adding the code '

R = R.to(I0.device) 
sigma = sigma.to(I0.device)

' inside this function can solve the issue.

**Comment by ConnorStoneAstro** [comment]
Hi @zhangxin702, 
I think this might be a PyTorch issue, not AstroPhot. PyTorch has some quirky behaviour compared to numpy since it can run code on CPU or GPU. Generally, any function that uses PyTorch will need to ensure all the input parameters are on the same device. So to use the `gaussian_torch` function, you would want to do something like this:

```
R = torch.tensor(5., device = "cuda", dtype = torch.float64)
sigma = torch.tensor(2., device = "cuda", dtype = torch.float64)
I0 = torch.tensor(1., device = "cuda", dtype = torch.float64)
result = gaussian_torch(R, sigma, I0)
```

In AstroPhot I have code like this all over the place, as well as a bunch of `.to(device, dtype)`  calls to ensure that everything stays on the right device and has the right datatype. If you are calling AstroPhot functions individually then you'll have to manually make sure everything is formatted properly.

I hope that helps! Let me know if there is anything else.

Best,
Connor

**Reply by zhangxin702** [reply]
Thank you for your answer.
